vendor/github.com/vmware/govmomi/object/host_certificate_info.go

```diff
@@ -86,10 +86,7 @@ func (info *HostCertificateInfo) FromURL(u *url.URL, config *tls.Config) error {
 
 	conn, err := tls.Dial("tcp", addr, config)
 	if err != nil {
-		switch err.(type) {
-		case x509.UnknownAuthorityError:
-		case x509.HostnameError:
-		default:
+		if !soap.IsCertificateUntrusted(err) {
```

Comment:
it would be helpful to add a release note describing the scenario where this cloud provider would not have been able to connect, or the error that would have surfaced